### PR TITLE
Add offline mode

### PR DESCRIPTION
**Pull request by jkodumal**

### CODE REVIEW

File: ldclient.go [hunk 1]
```diff
@@ -247,20 +249,41 @@ func (f Feature) Evaluate(user User) (interface{}, bool) {
...
 func (client *LDClient) Track(key string, user User, data interface{}) error {
+	if client.offline {
+		return nil
+	}
 	evt := newCustomEvent(key, user, data)
 	return client.processor.sendEvent(evt)
 }
 
+func (client *LDClient) SetOffline() {
+	client.offline = false
```
Comment (pkaeding) [inline ldclient.go]: looks like this logic (and `SetOnline`) is backwards.

**Comment by jkodumal** [comment]
Offline mode logic should be fixed now

File: ldclient_test.go [hunk 2]
```diff
@@ -1,9 +1,35 @@
...
 
+var config = Config{
+	BaseUri:       "https://localhost:3000",
+	Capacity:      1000,
+	FlushInterval: 5 * time.Second,
+	Logger:        log.New(os.Stderr, "[LaunchDarkly]", log.LstdFlags),
+	Timeout:       1500 * time.Millisecond,
+}
+
+func TestOfflineMOdeAlwaysReturnsDefaultValue(t *testing.T) {
```
Comment (pkaeding) [inline ldclient_test.go]: minor: MOde

**Comment by pkaeding** [comment]
lgtm

File: ldclient_test.go (hunk 2 above)
Comment (jkodumal) [inline ldclient_test.go]: Fixed in the user-rules branch

**Pull request merged**